src/common/services/api/order.service.spec.js

```diff
@@ -1206,4 +1208,159 @@ describe('order service', () => {
       expect(self.$window.localStorage.getItem('currentOrder')).toEqual('order id 2')
     })
   })
+
+  describe('submitOrder', () => {
+    let mockController;
+
+    beforeEach(() => {
+      mockController = {
+        submittingOrder: false,
+        onSubmittingOrder: jest.fn(),
+        onSubmitted: jest.fn(),
+        $scope: {
+          $emit: jest.fn(),
+        },
+      };
+
+      // Mock the submit() method to return a resolved observable
+      self.orderService.submit = jest.fn().mockReturnValue(Observable.of({}));
+    })
+
+    describe('another order submission in progress', () => {
+      it('should not submit the order twice', () => {
+        mockController.submittingOrder = true;
+        // Call submitOrder
+        const result = self.orderService.submitOrder(mockController)
+
+        // The submit method should not be called
+        expect(self.orderService.submit).not.toHaveBeenCalled();
+
+        // It should return an empty observable
+        expect(result).toEqual(Observable.empty());
+      })
+    })
+
+    describe('submit single order', () => {
+      beforeEach(() => {
+          self.orderService.clearCardSecurityCodes = jest.fn();
+          self.orderService.retrieveCardSecurityCode = jest.fn();
+          self.orderService.clearCoverFees = jest.fn();
+          mockController.loadCart = jest.fn();
+          self.$window.scrollTo = jest.fn();
+        })
+        
+        afterEach(() => {
+          expect(mockController.onSubmittingOrder).toHaveBeenCalledWith({ value: true })
+          expect(mockController.onSubmittingOrder).toHaveBeenCalledWith({ value: false })
+          expect(mockController.onSubmitted).toHaveBeenCalled()
+        })
+        
+        it('should submit the order normally if paying with a bank account', (done) => {
+          mockController.bankAccountPaymentDetails = {}
+          self.orderService.submitOrder(mockController).subscribe(
+            () => {
+              expect(self.orderService.submit).toHaveBeenCalled()
+              expect(self.orderService.clearCardSecurityCodes).toHaveBeenCalled()
+      
+              expect(mockController.$scope.$emit).toHaveBeenCalledWith(cartUpdatedEvent)
+              done()
+            })
+      })
+
+      it('should handle an error submitting an order with a bank account', (done) => {
+        self.orderService.submit.mockImplementation(() => Observable.throw({ data: 'error saving bank account' }))
+
+        mockController.bankAccountPaymentDetails = {}
+        self.orderService.submitOrder(mockController).subscribe(
+          () => {
+            fail()
+          },
```

Comment:
Some existing tests incorrectly use `fail`, but that is a mocha function, not a jest function. You can pass an error message to `done()` to get the test to fail and print the message.

```suggestion
          () => done("Observable unexpectedly succeeded"),
```